workers/data_refinery_workers/processors/create_compendia.py

```diff
@@ -0,0 +1,293 @@
+    job_context['merged_no_qn'] = untransposed_imputed_matrix_df
+
+    # Perform the Quantile Normalization
+    job_context = smasher._quantile_normalize(job_context, ks_check=False)
+    job_context['time_end'] = timezone.now()
+    job_context['formatted_command'] = "create_compendia.py"
+
+    return job_context
+
+
+def _create_result_objects(job_context: Dict) -> Dict:
+    """
+    Store and host the result as a ComputationalResult object.
+    """
+
+    result = ComputationalResult()
+    result.commands.append(" ".join(job_context['formatted_command']))
+    result.is_ccdl = True
+    result.is_public = True
+    result.time_start = job_context['time_start']
+    result.time_end = job_context['time_end']
+    try:
+        processor_key = "COMPENDIA"
+        result.processor = utils.find_processor(processor_key)
+    except Exception as e:
+        return utils.handle_processor_exception(job_context, processor_key, e)
+    result.save()
+
+    # Write the compendia dataframe to a file, overwriting the previous smash
+    job_context['merged_qn'].to_csv(job_context['smash_outfile'], sep='\t', encoding='utf-8')
+    compendia_tsv_computed_file = ComputedFile()
+    compendia_tsv_computed_file.absolute_file_path = job_context['smash_outfile']
+    compendia_tsv_computed_file.filename = job_context['smash_outfile'].split('/')[-1]
+    compendia_tsv_computed_file.calculate_sha1()
+    compendia_tsv_computed_file.calculate_size()
+    compendia_tsv_computed_file.is_smashable = False
+    compendia_tsv_computed_file.is_qn_target = False
+    compendia_tsv_computed_file.result = result
+    compendia_tsv_computed_file.save()
+
+    organism_key = list(job_context['samples'].keys())[0]
+    annotation = ComputationalResultAnnotation()
+    annotation.result = result
+    annotation.data = {
+        "organism_id": job_context['samples'][organism_key][0].organism_id,
+        "organism_name": job_context['samples'][organism_key][0].organism.name,
+        "is_qn": False,
+        "is_compendia": True,
+        "samples": [sample.accession_code for sample in job_context["samples"][organism_key]],
+        "num_samples": len(job_context["samples"][organism_key])
+    }
+    annotation.save()
+
+    # Save the related metadata file
+    metadata_computed_file = ComputedFile()
+    metadata_computed_file.absolute_file_path = job_context['metadata_tsv_paths'][0]
+    metadata_computed_file.filename = job_context['metadata_tsv_paths'][0].split('/')[-1]
+    metadata_computed_file.calculate_sha1()
+    metadata_computed_file.calculate_size()
+    metadata_computed_file.is_smashable = False
+    metadata_computed_file.is_qn_target = False
+    metadata_computed_file.result = result
+    metadata_computed_file.save()
+
+    # Create the resulting archive
+    final_zip_base = "/home/user/data_store/smashed/" + str(job_context["dataset"].pk) + "_compendia"
```

Comment:
Could we put the organism and compendia_version in the name of the archive file so that it's easy to tell while looking through s3?

Reply:
We do, it's below this code though